tests/docker-compose-storage.yml

```diff
@@ -0,0 +1,31 @@
+services:
+  minio:
+    image: minio/minio:RELEASE.2023-09-20T22-49-55Z
+    command: server /data --console-address ":9001"
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+      interval: 5s
+      timeout: 5s
+      retries: 10
+    ports:
+      - "9000:9000"
+      - "9001:9001"
+    environment:
+      MINIO_ROOT_USER: distribution
+      MINIO_ROOT_PASSWORD: password
+    volumes:
+      - ./miniodata/distribution:/data
+
+  minio-init:
+    image: minio/mc:RELEASE.2023-02-16T19-20-11Z
+    depends_on:
+      minio:
+        condition: service_healthy
+    entrypoint: >
+      /bin/bash -c "
+        /usr/bin/mc config host add minio http://minio:9000 $${MINIO_ROOT_USER} $${MINIO_ROOT_PASSWORD} && (
+          /usr/bin/mc stat minio/images-local || /usr/bin/mc mb minio/images-local
+        ) && /usr/bin/mc anonymous set public minio/images-local && /usr/bin/mc admin trace --path "
```

Comment:
Would it be worth creating Dockerfiles for these? (we can use build-arg for things we want to have customisable, such as the user and password, but perhaps that's not needed for this purpose. (maybe a `MINIO_VERSION=...` build-arg if you want to customise the version of the image that's used

Reply:
I think the only useful thing to justify the dedicated `Dockerfile` would be specifying the `min.io` version 🤔 but I'm willing to be persuaded otherwise 😄